addons/survey/tests/test_survey_security.py

```diff
@@ -145,17 +145,17 @@ def test_access_survey_survey_user(self):
         self.env['survey.question'].create({'title': 'My Page', 'sequence': 0, 'is_page': True, 'question_type': False, 'survey_id': survey.id})
         self.env['survey.question'].create({'title': 'My Question', 'sequence': 1, 'survey_id': survey.id})
 
-        # Read: all
+        # Read: restricted to self or no one
         surveys = self.env['survey.survey'].search([('title', 'ilike', 'Test')])
         self.assertEqual(surveys, self.survey | survey)
         surveys.read(['title'])
 
-        # Write: own only
-        survey.write({'title': 'New Title'})
+        # Write: restricted to self or no one
+        self.survey.write({'title': 'New Title', 'restrict_user_ids': [[4, self.survey.user_id.id]]})
```

Comment:
I find this `survey`vs `self.survey` quite confusing. Could we alias them as necessary to have `survey_1`, `survey_2` or `non_restricted_survey` and `restricted_users_survey`?